batavia/types/Str.js

```diff
@@ -519,6 +519,69 @@ Str.prototype.encode = function(encoding, errors) {
     }
 }
 
+Str.prototype.lstrip = function() {
+    var types = require('../types')
+
+    if (arguments.length === 0) {
+        return this.valueOf().trimLeft()
+    } else if (arguments.length === 1) {
+        var charsToTrim = arguments[0]
+        if (!types.isinstance(charsToTrim, [types.Str])) {
+            throw new exceptions.TypeError.$pyclass('lstrip arg must be None or str')
+        }
+        var result = this.valueOf()
+        while (charsToTrim.indexOf(result[0]) > -1) {
```

Comment:
maybe it would be better to count up, and then slice? could be a little more performant.
ie.:

```js
var i = 0;
while (charsToTrim.indexOf(result[i]) > -1) {
    i++;
}
result = result.slice(i);
```